DuckDuckGo.Fluent.Plugin/DuckResultFactory.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using static DuckDuckGo.Fluent.Plugin.JsonResult;
+using static DuckDuckGo.Fluent.Plugin.AppFunctions;
+
+namespace DuckDuckGo.Fluent.Plugin
+{
+    public class DuckResultFactory
+    {
+        private readonly DuckDuckGoApiResult _apiResult;
+        private readonly string _searchedText;
+
+        private DuckResultFactory(DuckDuckGoApiResult apiResult, string searchedText)
+        {
+            _apiResult = apiResult;
+            _searchedText = searchedText;
+        }
+
+        public static DuckResultFactory Create(DuckDuckGoApiResult root, string searchedText)
+        {
+            return new DuckResultFactory(root, searchedText);
+        }
+
+        public DuckResult GetAnswers()
+        {
+            if (string.IsNullOrWhiteSpace(_apiResult.Answer)) return null;
+            string resultType = _apiResult.AnswerType ?? "Answer";
+            return CreateDuckResult(_apiResult.Answer, resultType, null, ResultType.Answer, true);
+        }
+
+        public DuckResult GetDefinition()
+        {
+            return string.IsNullOrWhiteSpace(_apiResult.Definition)
+                ? null
+                : CreateDuckResult(_apiResult.Definition, "Define", _apiResult.DefinitionUrl, ResultType.Definition,
+                    true);
+        }
+
+        public DuckResult GetAbstract()
+        {
+            return string.IsNullOrWhiteSpace(_apiResult.AbstractText)
+                ? null
+                : CreateDuckResult(_apiResult.AbstractText, "Abstract", _apiResult.AbstractUrl, ResultType.Abstract,
+                    true);
+        }
+
+        public List<DuckResult> GetRelatedTopics()
```

Comment:
Instead of returning a `List` I recommend returning `IEnumerable<DuckResult>` and then using `yield return`s to return items one by one.
This way the consumer (the search app) can decide if he needs to allocate a `List` or not (I think it's not needed)

Reply:
Also it would help with returning results faster, since the consumer won't need to wait for the whole list to be filled

Reply:
All `List`s have been converted to `IEnumerable` now.